File: CgmLogger/src/main/java/com/kludgenics/cgmlogger/data/logdata/location/api/GeoApi.java
package com.kludgenics.cgmlogger.data.logdata.location.api;

import com.kludgenics.cgmlogger.data.logdata.location.data.GeocodedLocation;
import com.kludgenics.cgmlogger.data.logdata.location.data.Position;
import rx.Observable;

/**
 * Created by matthiasgranberry on 5/11/15.
 */
public interface GeoApi {
    Observable<GeocodedLocation> getCurrentLocation();
    Observable<GeocodedLocation> getCurrentLocation(String categories);
    Observable<GeocodedLocation> search(Position position);
    Observable<GeocodedLocation> search(Position position, String categories);
    Observable<AutoCompleteResult> autoComplete(Position position, String query);
    Observable<AutoCompleteResult> autoComplete(Position position, String query, String categories);
    Observable<GeocodedLocation> getInfo(String id);
}

File: CgmLogger/src/main/java/com/kludgenics/cgmlogger/data/location/GeoApi.kt
package com.kludgenics.cgmlogger.data.location

import com.kludgenics.cgmlogger.data.location.places.AutoCompleteResult
import com.kludgenics.cgmlogger.data.location.data.GeocodedLocation
import com.kludgenics.cgmlogger.data.location.data.Position
import rx.Observable

/**
 * Created by matthiasgranberry on 5/11/15.
 */
public interface GeoApi {
    public fun getCurrentLocation(): Observable<GeocodedLocation>
    public fun getCurrentLocation(categories: String): Observable<GeocodedLocation>
    public fun search(position: Position): Observable<GeocodedLocation>
    public fun search(position: Position, categories: String?): Observable<GeocodedLocation>
    public fun autoComplete(position: Position, query: String?): Observable<AutoCompleteResult>
    public fun autoComplete(position: Position, query: String?, categories: String?): Observable<AutoCompleteResult>
    public fun getInfo(id: String): Observable<GeocodedLocation>
}
